tests/test_integration.py

```diff
@@ -1338,3 +1339,30 @@ def test_issues_539_abi3_resolution():
     res.assert_success()
 
     subprocess.check_call([cryptography_pex, '-c', 'import cryptography'])
+
+
+@pytest.mark.skip("Acceptance test for landing https://github.com/pantsbuild/pex/issues/716.")
+def test_reproducible_build():
+
+  def assert_reproducible(args):
+    with temporary_dir() as td:
+      pex1 = os.path.join(td, '1.pex')
+      pex2 = os.path.join(td, '2.pex')
+      run_pex_command(args + ['-o', pex1])
+      run_pex_command(args + ['-o', pex2])
+      assert filecmp.cmp(pex1, pex2, shallow=False)
```

Comment:
I think that you'll want to maybe have two asserts: one that checks the unzipped pex, then another that checks the zipped pex. I think that the former might have more useful/readable output while being easier to pass, while the latter is harder to pass, and has less useful output.

Reply:
Good suggestion! Added. Think this is the appropriate level of testing now—original was nearly impossible to debug and then anything more than this would be getting too specific and complicated.